main.rb

```diff
@@ -0,0 +1,205 @@
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    iarr = to_a
+    size = iarr.length
+    x = 0
+    until x == size
+      yield(iarr[x])
+      x += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum :my_each_with_index unless block_given?
+
+    iarr = self
+    size = iarr.length
+    x = 0
+    until x == size
+      yield(iarr[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    varr = []
+    iarr = self
+    iarr.my_each { |num| varr << num if yield(num) == true }
+    varr
+  end
+
+  def my_all?(val = nil)
+    iarr = self
+    status = true
+    if block_given? && val.nil?
+      iarr.my_each { |num| status = false unless yield(num) }
+    elsif val
+      status = true
+      iarr.my_each do |num|
+        status = true if num == val && status == true
+        status = false if num != val && status == true
+      end
+    else
+      status = true
+      iarr.my_each do |num|
+        status = true if num && num != false && status == true
+        status = false if (num.nil? || num == false) && status == true
+      end
+    end
+    status
+  end
+
+  def my_any?(val = nil)
+    iarr = self
+    status = false
+    if block_given? && val.nil?
+      iarr.my_each { |num| status = true if yield(num) }
+    elsif val
+      status = false
+      iarr.my_each do |num|
+        status = true if num == val && status == false
+      end
+    else
+      status = false
+      iarr.my_each do |num|
+        status = true if num && status == false
+      end
+    end
+    status
+  end
+
+  def my_none?(val = nil)
+    status = true
+    iarr = self
+    if block_given?
+      iarr.my_each do |num|
+        status = false if yield(num)
+      end
+    elsif val.is_a? Class
+      iarr.my_each do |num|
+        status = false if num.is_a? val
+      end
+    elsif val
+      iarr.my_each do |num|
+        status = false if val == num
+      end
+    elsif val.nil? && !block_given?
+      iarr.my_each do |num|
+        status = false if num
+      end
+    end
+    status
+  end
```

Comment:
[Optional]

See the previous comment on handling cases where the argument is a `Regex` or a `Class`.

### Find more details about [`none?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F)